### PR TITLE
Add interactive first-run onboarding flow

### PR DESCRIPTION
**Pull request by Emanuele-web04**
## What Changed

- Added a six-step first-run onboarding dialog covering welcome, feature tour, providers, theme, project setup, and completion.
- Added provider detection, enable/disable controls, inline sign-in terminals, project creation, and appearance setup.
- Added persistent server-backed onboarding completion state with a local fallback.
- Added reusable onboarding logic and focused unit tests for navigation, gating, provider classification, summaries, and selection toggling.
- Added a Settings entry point to replay onboarding.

## Why

New installations now receive guided setup before entering the main workspace. The flow helps users configure available agents, choose an appearance, add an initial project, and understand key Synara capabilities while preserving safe behavior across reloads and server reconfiguration.

## UI Changes

- Added the interactive onboarding modal and progress footer.
- Added feature-tour tabs, provider setup status, inline authentication terminals, project folder selection, and completion summary.
- Screenshots and interaction video are not included in this draft.

## Verification

- Added focused tests in `apps/web/src/onboarding/logic.test.ts`.
- After the Cursor findings were repaired, all 23 focused onboarding logic tests passed. The first regression run had 4 failures on the pre-fix source.
- Both hook-level browser regressions passed: a config refetch failure preserves the intro when identity is cached; completing a Settings replay before config arrives stays dismissed, while changing installations still opens the new tour. Both tests failed before the second fix.
- Local browser fixtures mock config/settings responses; live provider sign-in was not exercised locally. Browser runs emitted React act-environment warnings despite passing assertions.
- Current-head Cursor review and complete hosted CI must pass before merge.

## Checklist

- [x] This PR is small and focused
- [x] I explained what changed and why
- [ ] I included before/after screenshots for any UI changes
- [ ] I included a video for animation/interaction changes

---

> [!NOTE]
> **Medium Risk**
> Touches first-launch UX, server settings persistence, and provider sign-in terminals scoped away from real threads; incorrect gating could hide or replay onboarding across installations.
> 
> **Overview**
> Adds a **six-step welcome tour** (intro, feature tabs, agent setup, appearance, first project, done) mounted lazily from the root route, with a Settings control to **replay** it and search indexing for that entry.
> 
> **Completion is persisted** via new server field `onboardingCompletedAt` (contracts + app settings sync/patch), plus an installation-scoped local fallback keyed off `worktreesDir`. Pure gate/reconcile logic decides when to show the tour (fresh install with no ordinary projects), auto-dismiss before the user engages if later data proves the install is already configured, and backfills the server marker for legacy installs or failed writes. **Restore defaults** no longer clears the onboarding timestamp so setup does not replay.
> 
> The tour includes provider detection, enable/disable with optimistic drafts, **inline sign-in terminals** (isolated synthetic thread scope), live theme picking, and project add via shared create/recover. **AppSnap’s welcome sheet** waits on the onboarding startup gate so modals do not stack. Folder drop handling is **factored** into `folderDrop` + `useWindowFolderDrop` (Create Project dialog and onboarding). Browser fixtures stub `serverGetSettings` with completion set so existing UI tests stay unobstructed; unit/browser tests cover gate logic and installation identity edge cases. Provider metadata gains **setup docs links** for missing runtimes.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 8544b938a0dcb6f0761f175b46f4b7b57e170545. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: apps/web/src/onboarding/steps/ProvidersStep.tsx [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+              <div className="flex items-center gap-3 px-2 py-2">
+                <Checkbox
+                  checked={enabled}
+                  aria-label={`${enabled ? "Disable" : "Enable"} ${descriptor.displayName}`}
+                  onCheckedChange={(checked) =>
+                    updateSettings({
+                      disabledProviders: setDisabled(
+                        settings.disabledProviders,
+                        descriptor.kind,
+                        !Boolean(checked),
```
[the remark this answers is not in the document]
Comment (Emanuele-web04) [reply apps/web/src/onboarding/steps/ProvidersStep.tsx]: Fixed in 44d0430: writes derive from an optimistic local draft, and in ebb070075 the pending-write count moved into state so a rejected write resyncs the draft from the server.

File: apps/web/src/onboarding/useOnboarding.ts [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    serverCompletedAt,
+    localCompletedAt,
+  ]);
+
+  const complete = () => {
+    const completedAt = new Date().toISOString();
+    setStorage({ completedAt });
+    if (settings.onboardingCompletedAt === null) {
+      updateSettings({ onboardingCompletedAt: completedAt });
+    }
```
[the remark this answers is not in the document]
Comment (Emanuele-web04) [reply apps/web/src/onboarding/useOnboarding.ts]: Fixed in 44d0430: completion is written locally first and reconciled to the server on later launches if the write failed.

File: apps/web/src/appSettings.ts [hunk 3]
```diff
@@ -254,6 +254,8 @@ const PersistedHiddenModels = Schema.Array(
 
 export const AppSettingsSchema = Schema.Struct({
   claudeBinaryPath: Schema.String.check(Schema.isMaxLength(4096)).pipe(withDefaults(() => "")),
+  // Server-backed first-run marker; see ServerSettings.onboardingCompletedAt.
+  onboardingCompletedAt: Schema.NullOr(Schema.String).pipe(withDefaults((): string | null => null)),
```
[the remark this answers is not in the document]
Comment (Emanuele-web04) [reply apps/web/src/appSettings.ts:258]: Fixed in 44d0430: resetSettings strips onboardingCompletedAt from the patch and preserves it locally.

File: apps/web/src/onboarding/useOnboarding.ts [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+      settingsAvailable: settingsQuery.isSuccess,
+      projectCount,
+      serverCompletedAt,
+      localCompletedAt,
+    });
+    if (resolved === "pending") {
+      return;
+    }
+    latchedRef.current = true;
+    setGate(resolved);
```
[the remark this answers is not in the document]
Comment (Emanuele-web04) [reply apps/web/src/onboarding/useOnboarding.ts]: Fixed in 44d0430: the gate is re-evaluated live instead of latched; a first-run open is revised back to closed by later authoritative data unless the user already reached a setup step.

File: apps/web/src/onboarding/logic.ts [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+ * marker only covers the case where server settings are unreachable.
+ */
+export function resolveOnboardingGate(input: OnboardingGateInputs): OnboardingGate {
+  if (!input.threadsHydrated || !input.settingsSettled) {
+    return "pending";
+  }
+  const alreadyCompleted = input.settingsAvailable
+    ? input.serverCompletedAt !== null
+    : input.localCompletedAt !== null;
+  return !alreadyCompleted && input.projectCount === 0 ? "show" : "hidden";
```
[the remark this answers is not in the document]
Comment (Emanuele-web04) [reply apps/web/src/onboarding/logic.ts]: Fixed in 44d0430: installs with projects and no marker get a server marker written once.

File: apps/web/src/routes/__root.tsx [hunk 6]
```diff
@@ -29,6 +29,8 @@ import { DesktopWindowControls } from "../components/DesktopWindowControls";
 import { RunningChatsQuitCoordinator } from "../components/RunningChatsQuitCoordinator";
 import { AppSnapCoordinator } from "../components/AppSnapCoordinator";
 import { AppSnapWelcomeDialog } from "../components/AppSnapWelcomeDialog";
+import { OnboardingDialog } from "../onboarding/OnboardingDialog";
```
[the remark this answers is not in the document]
Comment (Emanuele-web04) [reply apps/web/src/routes/__root.tsx]: Fixed in 44d0430: the root only imports the gate hook; the dialog (and xterm behind it) loads via React.lazy the first time the tour opens.

File: apps/web/src/onboarding/steps/ProvidersStep.tsx [hunk 7]
```diff
@@ -0,0 +1,243 @@
...
+      next.add(provider);
+    } else {
+      next.delete(provider);
+    }
+    draftRef.current = next;
+    setDraft(next);
+    pendingWritesRef.current += 1;
+    void updateSettingsAndWait({ disabledProviders: [...next] }).finally(() => {
+      pendingWritesRef.current -= 1;
+    });
```
[the remark this answers is not in the document]
Comment (Emanuele-web04) [reply apps/web/src/onboarding/steps/ProvidersStep.tsx:107]: Fixed in ebb070075: the pending-write count is React state, so the resync effect reruns when the final write settles.

File: apps/web/src/onboarding/logic.ts [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+ *
+ * The result is *not* authoritative until the inputs are: callers re-evaluate as
+ * snapshots and settings arrive instead of latching the first non-pending answer.
+ */
+export function resolveOnboardingGate(input: OnboardingGateInputs): OnboardingGate {
+  if (!input.threadsHydrated || !input.settingsSettled) {
+    return "pending";
+  }
+  const completed = input.serverCompletedAt !== null || input.localCompletedAt !== null;
+  return !completed && input.projectCount === 0 ? "show" : "hidden";
```
[the remark this answers is not in the document]
Comment (Emanuele-web04) [reply apps/web/src/onboarding/logic.ts:96]: Fixed in ebb070075: the local marker records the server's worktreesDir as an installation key and only counts (or reconciles) against the same installation.

File: apps/web/src/routes/__root.tsx [hunk 9]
```diff
@@ -307,6 +317,7 @@ function RootRouteView() {
           <TaskCompletionNotifications />
           <QueuedComposerDrainCoordinator />
           <AppSnapWelcomeDialog />
+          <GlobalOnboardingDialog />
```
[the remark this answers is not in the document]
Comment (Emanuele-web04) [reply apps/web/src/routes/__root.tsx:320]: Fixed in ebb070075: AppSnapWelcomeDialog stays closed until the onboarding gate has settled and the tour is closed.

File: apps/web/src/onboarding/onboardingTerminalScope.ts [hunk 10]
```diff
@@ -0,0 +1,12 @@
...
+// Purpose: Synthetic terminal scope ids for the provider sign-in terminals shown during
+//          the welcome tour, so they never collide with a real thread's terminal state.
+// Layer: Web domain helper
+
+import type { ProviderKind, ThreadId } from "@synara/contracts";
+
+export const ONBOARDING_TERMINAL_SCOPE_PREFIX = "onboarding-terminal:";
+
+export function onboardingTerminalThreadId(provider: ProviderKind): ThreadId {
+  return `${ONBOARDING_TERMINAL_SCOPE_PREFIX}${provider}` as ThreadId;
```
[the remark this answers is not in the document]
Comment (Emanuele-web04) [reply apps/web/src/onboarding/onboardingTerminalScope.ts]: Fixed in ebb070075: the synthetic scope id includes a per-page nonce.

File: apps/web/src/onboarding/steps/ProvidersStep.tsx [hunk 11]
```diff
@@ -0,0 +1,261 @@
...
+      <DisclosureRegion open={connecting !== undefined}>
+        {connecting && connectingSignInCommand !== undefined && homeDir !== null ? (
+          <ProviderConnectTerminal
+            key={connecting.descriptor.kind}
+            provider={connecting.descriptor.kind}
+            signInCommand={connectingSignInCommand}
+            cwd={homeDir}
+          />
+        ) : null}
+      </DisclosureRegion>
```
[the remark this answers is not in the document]
Comment (Emanuele-web04) [reply apps/web/src/onboarding/steps/ProvidersStep.tsx:293]: Fixed in ebb070075: the terminal region scrolls into view on open and has its own always-visible Done control, independent of the card's needs-sign-in state.

File: apps/web/src/onboarding/steps/ProjectStep.tsx [hunk 12]
```diff
@@ -0,0 +1,192 @@
...
+      setSubmitting(false);
+    }
+  };
+
+  const browse = async () => {
+    const api = readNativeApi();
+    if (!api) return;
+    setPicking(true);
+    try {
+      const picked = await api.dialogs.pickFolder();
```
[the remark this answers is not in the document]
Comment (Emanuele-web04) [reply apps/web/src/onboarding/steps/ProjectStep.tsx:90]: Fixed in ebb070075: pickFolder rejections populate the step's error state.

File: apps/web/src/onboarding/steps/ProvidersStep.tsx [hunk 13]
```diff
@@ -0,0 +1,261 @@
...
+  const refreshProviderStatuses = useRefreshProviderStatusesNow();
+  const homeDir = useWorkspacePathsStore((store) => store.homeDir);
+  const [connectingProvider, setConnectingProvider] = useState<ProviderKind | null>(null);
+  const [refreshing, setRefreshing] = useState(false);
+  const { disabled: disabledSet, setProviderDisabled } = useDisabledProvidersDraft();
+
+  const refresh = async () => {
+    setRefreshing(true);
+    try {
+      await refreshProviderStatuses({ silent: true });
```
[the remark this answers is not in the document]
Comment (Emanuele-web04) [reply apps/web/src/onboarding/steps/ProvidersStep.tsx]: Fixed in ebb070075: manual Re-detect no longer passes silent; the entry probe and the post-sign-in refresh stay silent.

File: apps/web/src/onboarding/steps/ThemeStep.tsx [hunk 14]
```diff
@@ -0,0 +1,108 @@
...
+
+export function ThemeStep() {
+  const { theme, setTheme, activeTheme, resolvedTheme, setCodeThemeId } = useTheme();
+  const selectedPackId = activeTheme.codeThemeId;
+  const selectPack = (codeThemeId: string) => {
+    for (const variant of THEME_VARIANTS) setCodeThemeId(variant, codeThemeId);
+  };
+  const radioItemProps = useRadioGroupKeyboardNav({
+    values: ONBOARDING_THEME_PACK_IDS,
+    value: selectedPackId,
```
[the remark this answers is not in the document]
Comment (Emanuele-web04) [reply apps/web/src/onboarding/steps/ThemeStep.tsx]: Fixed in ebb070075: the roving tab stop falls back to the first listed pack when the active pack is not in the list.

File: apps/web/src/onboarding/OnboardingDialog.tsx [hunk 15]
```diff
@@ -0,0 +1,210 @@
...
+        return { label: "Get started", onPrimary: goNext };
+      case "tour":
+        return { label: "Set up", onPrimary: goNext };
+      case "providers":
+      case "theme":
+        return { label: "Continue", onPrimary: goNext };
+      case "project":
+        return projectResults.length > 0
+          ? { label: "Continue", onPrimary: goNext }
+          : { label: "Skip for now", onPrimary: goNext };
```
[the remark this answers is not in the document]
Comment (Emanuele-web04) [reply apps/web/src/onboarding/OnboardingDialog.tsx:129]: Fixed in ebb070075: the project step reports busy; the footer locks Back/Skip/primary and the dialog refuses to close until creation settles.
